File: main/src/main/java/org/anibyl/slounik/network/ArticlesInfo.java
package org.anibyl.slounik.network;

import org.anibyl.slounik.Article;

import java.util.ArrayList;

/**
 * Represents article information.
 *
 * Contains articles and connected information.
 *
 * Created by Usievaład Kimajeŭ on 29.4.15 16.08.
 */
public class ArticlesInfo {
    public enum Status {
        SUCCESS,
        IN_PROCESS,
        FAILURE
    }

    private ArrayList<Article> articles;
    private Status status;

    public ArticlesInfo(ArrayList<Article> articles, Status status) {
        this.articles = articles;
        this.status = status;
    }

    public ArticlesInfo(ArrayList<Article> articles) {
        this.articles = articles;

        if (articles == null) {
            status = Status.FAILURE;
        } else {
            status = Status.SUCCESS;
        }
    }

    public ArticlesInfo(Status status) {
        this.status = status;
    }

    public ArrayList<Article> getArticles() {
        return articles;
    }

    public Status getStatus() {
        return status;
    }
}

File: main/src/main/java/org/anibyl/slounik/network/ArticlesInfo.kt
package org.anibyl.slounik.network

/**
 * Represents article information.

 * Contains articles and connected information.

 * @author Usievaład Kimajeŭ
 * @created 29.04.2015
 */
class ArticlesInfo {
	enum class Status {
		SUCCESS,
		IN_PROCESS,
		FAILURE
	}

	var articles: List<Article>? = null
		private set
	var status: Status? = null
		internal set

	constructor(articles: List<Article>?, status: Status) {
		this.articles = articles
		this.status = status
	}

	constructor(articles: List<Article>?) {
		this.articles = articles

		if (articles == null) {
			status = Status.FAILURE
		} else {
			status = Status.SUCCESS
		}
	}

	constructor(status: Status) {
		this.status = status
	}
}
